File: 0547-number-of-provinces/0547-number-of-provinces.java
class Solution {
    boolean[] visited;
    int n;
    public int findCircleNum(int[][] isConnected) {
        n = isConnected.length;
        visited = new boolean[isConnected.length];
        int count = 0;
        for(int i = 0; i < n; i++) {
            if(!visited[i]) {
                count++;
                dfs(i,isConnected);
            }
        }
        return count;
    }
    
    private void dfs(int index, int[][] isConnected) {
        
        if(visited[index]) return;
        visited[index] = true;
        for(int j = 0; j < n; j++) {
            if(isConnected[index][j] == 1) dfs(j,isConnected);
        }
        
    }
}

File: 0547-number-of-provinces/0547-number-of-provinces.js
/**
 * @param {number[][]} isConnected
 * @return {number}
 */
var findCircleNum = function(isConnected) {
    const visited = new Array(isConnected.length).fill(false);
    const dfs = (node) => {
        if(visited[node]) return;
        visited[node] = true;

        for(let i = 0; i < isConnected[node].length; i++) {
            if(isConnected[node][i] === 1)
                dfs(i);
        }
    }

    let res = 0;
    for(let i = 0 ; i < isConnected.length; i++) {
        if(!visited[i]) {
            res++;
            dfs(i);
        }
    }

    return res;
};